File: leetcode/python/70.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1 or n == 2:
            return n

        l = [0 for i in range(n + 1)]
        l[1] = 1
        l[2] = 2

        for i in range(3, n + 1):
            l[i] = l[i - 1] + l[i - 2]

        return l[-1]

File: leetcode/java/70.java
class Solution {
  public int climbStairs(int n) {
    int[] arr = new int[46];

    arr[0] = 0;
    arr[1] = 1;
    arr[2] = 2;

    for (int i = 3; i <= n; i++) {
      arr[i] = arr[i - 1] + arr[i - 2];
    }

    return arr[n];
  }
}
